srcs/execution/builtins/opt.c

```diff
@@ -0,0 +1,33 @@
+#include "opt.h"
+#include "builtins.h"
+
+/*
+** could be use as generic f() for other builtins
+** check options presence:
+** -e, -E, -s, -n, -esn
+*/
+
+int		retrieve_options(int argc, char **argv, t_opt *o)
+{
+	int	i;
+	int	j;
+
+	i = 0;
+	o->s = false;
+	o->n = false;
+	o->e = false;
+	while (++i < argc && argv[i][0] == '-' && argv[i][1] != '-')
+	{
+		j = 0;
+		while (argv[i][++j] != '\0')
+		{
+			if (argv[i][j] == 's')
+				o->s = true;
+			else if (argv[i][j] == 'n')
+				o->n = true;
+			else if (argv[i][j] == 'e')
+				o->e = true;
```

Comment:
Thanks for the fix proposition.
I planned to use this function as a generic one for `env -i` for instance.
To do so, I planned to `-i` on the structure and on the f().
Finally, `echo -i` won't display `-i`.

I found a solution.